### PR TITLE
ci(NODE-6501): use latest node in spec benchmarks

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

We use latest LTS (v22.11.0) for performance testing.  Also, the runner now has a max number of executions set to 10 million, to prevent OOM errors on the returnDocument test.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/config.in.yml [hunk 1]
```diff
@@ -1157,7 +1157,7 @@ tasks:
         type: setup
         params:
           updates:
-            - { key: NODE_LTS_VERSION, value: v18.16.0 }
+            - { key: NODE_LTS_VERSION, value: v22.11.0 }
             - { key: NPM_VERSION, value: "9" }
```
Comment (nbbeeken) [inline .evergreen/config.in.yml:1161]: We don't need the npm pin anymore either

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: test/benchmarks/mongoBench/runner.js [hunk 2]
```diff
@@ -162,12 +163,17 @@ class Runner {
...
     let count = 1;
 
     const taskTimer = benchmark._taskType === 'sync' ? timeSyncTask : timeAsyncTask;
 
-    while (time < maxExecutionTime && (time < minExecutionTime || count < minExecutionCount)) {
+    while (
+      time < maxExecutionTime &&
+      (time < minExecutionTime || count < minExecutionCount) &&
+      count < maxExecutionAttempts
+    ) {
```
Comment (nbbeeken) [inline test/benchmarks/mongoBench/runner.js:176]:
```js
 _errorHandler(error) {
    let currentError = error;
    while (currentError) {
      this.reporter(
        `${currentError !== error ? 'Caused by' : 'Error'}: ${currentError.name} - ${currentError.message} - ${currentError.stack}`
      );
      currentError = currentError.cause;
    }
    throw error;
  }
```

I don't feel too strongly about the print logic, it was just to remove all those ids from the bulk error, but can we make errors throw now? I don't think we should have the code "tolerate" errors and continue and ruin the entire calculation with `NaN` anymore.

File: test/benchmarks/mongoBench/runner.js [hunk 3]
```diff
@@ -57,6 +57,7 @@ class Runner {
     this.minExecutionTime = options.minExecutionTime || CONSTANTS.DEFAULT_MIN_EXECUTION_TIME;
```
Comment (nbbeeken) [inline test/benchmarks/mongoBench/runner.js:57]:
```js
const sortedData = [].concat(rawData).sort();
```

I cannot comment on the line directly. This does not sort the values correctly

```js
[].concat([1.2, 10.1, 2.4, 20.2]).sort();
[ 1.2, 10.1, 2.4, 20.2 ]
```
10 doesn't come before 2 😢

File: test/benchmarks/mongoBench/runner.js [hunk 4]
```diff
@@ -162,12 +163,17 @@ class Runner {
     const minExecutionCount = this.minExecutionCount;
     const minExecutionTime = this.minExecutionTime;
     const maxExecutionTime = this.maxExecutionTime;
+    const maxExecutionAttempts = this.minExecutionCount;
```
Comment (nbbeeken) [inline test/benchmarks/mongoBench/runner.js:166]:
```suggestion
    const maxExecutionAttempts = this.maxExecutionCount;
```

🍝

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: test/benchmarks/mongoBench/runner.js [hunk 5]
```diff
@@ -162,12 +163,17 @@ class Runner {
...
     let time = performance.now() - start;
     let count = 1;
 
     const taskTimer = benchmark._taskType === 'sync' ? timeSyncTask : timeAsyncTask;
 
-    while (time < maxExecutionTime && (time < minExecutionTime || count < minExecutionCount)) {
+    while (
```
Comment (nbbeeken) [inline test/benchmarks/mongoBench/runner.js:172]: Can't comment on the line. Inside `_runSuite`:

```js
this.reporter(` Executed Benchmark "${name}" =`, result[name]);
```

**Review by nbbeeken — APPROVED** [review]

File: test/benchmarks/mongoBench/runner.js (hunk 3 above)
Comment (dariakp) [reply test/benchmarks/mongoBench/runner.js:57]: It does in the alphabet 😿

File: test/benchmarks/mongoBench/runner.js [hunk 6]
```diff
@@ -39,7 +39,7 @@ function calculateMicroBench(benchmark, data) {
   const rawData = data.rawData;
   const count = data.count;
 
-  const sortedData = [].concat(rawData).sort();
+  const sortedData = [].concat(rawData).sort((a, b) => a - b);
```
Comment (nbbeeken) [inline test/benchmarks/mongoBench/runner.js:42]: One last fix, can we calculate the median properly? average of middle two if even and the middle value if odd. (we could delete the other percentiles here, we don't use them)

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: test/benchmarks/mongoBench/runner.js (hunk 6 above)
Comment (nbbeeken) [reply test/benchmarks/mongoBench/runner.js:42]: https://github.com/mongodb/specifications/blob/master/source/benchmarking/benchmarking.md#:~:text=For%20each%20micro,for%20every%20driver.

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**